validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -445,13 +453,19 @@ def start(self):
         super().start()
 
     def run(self):
+        has_chain_head = self._current_chain_head_func is not None
         while not self._stopped:
             try:
                 if self._peering_mode == 'dynamic':
                     self.retry_dynamic_peering()
                 elif self._peering_mode == 'static':
                     self.retry_static_peering()
 
+                has_chain_head = has_chain_head or \
```

Comment:
I think this would be clearer as
```
if not has_chain_head:
    has_chain_head = self._current_chain_head_func() is not None
```

Reply:
In that case, you need a second if block to react to the chain head still being not none, since it being set is an asynchronous operation - I.e. it may have been reset.  Using the or is equivalent, and makes the operation extremely cheap for all subsequent checks (no trips to the block store).